### DISCUSSION TITLE
How to support H265 in frontend?

### DISCUSSION THREAD

**Discussion by quanqigu** [pion/webrtc#1879] [Q&A]
Is there an easy way to support H265 in frontend?
Our camera can generate H265 stream, we want use Pion to push this h265 RTP stream to ION-SFU, and then frontend can get the camera stream from SFU. 

But I don't know how to support it in browser. Is there any way to do this?

I think if we can wrap ffmpeg in WASM and transcode rtp received from RtpReceiver to h264 and then play it?

We need store the H265 video of camera to the disk, and we don't want transcode in server as it need much resource, we can't afford that cost.

**Comment by Sean-Der** [comment]
The only browser that supports H265 today is Safari. You could register H265 in the MediaEngine and Write/Read RTP packets.

We also can support read/writing Samples after [this](https://github.com/pion/rtp/pull/134/files) is merged. I will work on getting this merged into master.

**Reply by quanqigu** [reply]
Wow, that's great. I am looking forward to this PR. 
Thank you very much for your great job, great project.

**Comment by kalmastenitin** [comment]
```rtpSender.Read()``` 
 doesnot work for H265 feed. hence unable to send stream
